sample-resource-plugin/src/main/java/org/opensearch/sample/resource/actions/transport/SearchResourceTransportAction.java

```diff
@@ -18,99 +16,30 @@
-    private final ResourceSharingClient resourceSharingClient;
+    private final PluginClient pluginClient;
 
     @Inject
-    public SearchResourceTransportAction(TransportService transportService, ActionFilters actionFilters, Client nodeClient) {
+    public SearchResourceTransportAction(TransportService transportService, ActionFilters actionFilters, PluginClient pluginClient) {
         super(SearchResourceAction.NAME, transportService, actionFilters, SearchRequest::new);
-        this.transportService = transportService;
-        this.nodeClient = nodeClient;
-        this.resourceSharingClient = ResourceSharingClientAccessor.getInstance().getResourceSharingClient();
+        this.pluginClient = pluginClient;
     }
 
     @Override
     protected void doExecute(Task task, SearchRequest request, ActionListener<SearchResponse> listener) {
-        ThreadContext threadContext = transportService.getThreadPool().getThreadContext();
-        try (ThreadContext.StoredContext ignore = threadContext.stashContext()) {
-            if (resourceSharingClient == null) {
-                nodeClient.search(request, listener);
-                return;
-            }
-            // if the resource sharing feature is enabled, we only allow search from documents that requested user has access to
-            searchFilteredIds(request, listener);
-        } catch (Exception e) {
-            log.error("Failed to search resources", e);
-            listener.onFailure(e);
-        }
-    }
-
-    private void searchFilteredIds(SearchRequest request, ActionListener<SearchResponse> listener) {
-        SearchSourceBuilder src = request.source() != null ? request.source() : new SearchSourceBuilder();
-        ActionListener<Set<String>> idsListener = ActionListener.wrap(resourceIds -> {
-            mergeAccessibleFilter(src, resourceIds);
-            request.source(src);
-            nodeClient.search(request, listener);
-        }, e -> {
-            mergeAccessibleFilter(src, Set.of());
-            request.source(src);
-            nodeClient.search(request, listener);
-        });
-
-        resourceSharingClient.getAccessibleResourceIds(RESOURCE_INDEX_NAME, idsListener);
-    }
-
-    private void mergeAccessibleFilter(SearchSourceBuilder src, Set<String> resourceIds) {
-        QueryBuilder accessQB;
-
-        if (resourceIds == null || resourceIds.isEmpty()) {
-            // match nothing
-            accessQB = QueryBuilders.boolQuery().mustNot(QueryBuilders.matchAllQuery());
-        } else {
-            // match only from a provided set of resources
-            accessQB = QueryBuilders.idsQuery().addIds(resourceIds.toArray(new String[0]));
-        }
-
-        QueryBuilder existing = src.query();
-        if (existing == null) {
-            // No existing query → just the filter
-            src.query(QueryBuilders.boolQuery().filter(accessQB));
-            return;
-        }
+        pluginClient.search(request, listener);
```

Comment:
we need to update [plugin-dev documentation](https://github.com/opensearch-project/security/blob/main/RESOURCE_SHARING_AND_ACCESS_CONTROL.md#plugin-implementation-requirements) to declare plugin as IdentityAwarePlugin.